### DISCUSSION TITLE
How to retrieve all orders using GetOrders()

### DISCUSSION THREAD

**Discussion by olsw** [amzn/selling-partner-api-models#3109] [General] [answered]
We normally only have < 100 orders in a day and our request to GetOrders works perfectly. We have had an unusually busy day so having to make use of NextToken to pull the next set of orders. However, we are finding that every response has NextToken present and meaning we are getting an endless number of responses even when setting MarketPlaceId and the time for only a single day

Has anyone else come across this issue?

We are putting in an arbitrary stop after e.g. 20 times using a NextToken value but I'm thinking I am either using it incorrectly or there is a bug somewhere

**Comment by teddy-codes** [comment]
You definitely should use issues for this. You would want to get orders and check the timestamps of the last order. An arbitrary stop is very wasteful. You should store the value of the last stop time that you had processed and go through that again to ensure that you are not entering duplicates.

If you have your own DB of orders, then you could also go the route of having the order ID as unique and inserting as you are getting the pages. The first time that you get a duplicate order id, you should stop requesting the next token.

**Comment by olsw** [comment, marked as answer]
Just plain old bad code

**Reply by teddy-codes** [reply]
what is? 

The API was rushed and is trash. We as MWS devs are just getting pushed by Amazon to use this trash.

**Reply by olsw** [reply]
In this instance, my code!

**Reply by teddy-codes** [reply]
lol! It's all our code bad?
